drivers/include/periph/flashsector.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (C) 2017 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_flashsector Flash sector driver
+ * @ingroup     drivers_periph
+ * @brief       Low-level flash sector interface
+ *
+ * @{
+ * @file
+ * @brief       Low-level flash sector peripheral driver interface
+ *
+ * This interface provides a very simple and straight forward way for writing
+ * a MCU's internal flash. This interface is only capable of reading, verifying,
+ * and writing complete flash sectors, it has no support for partial flash
+ * access.
+ * This enables for very slim and efficient implementations.
+ *
+ * A module for more fine-grained access of memory locations can easily be
+ * programmed on top of this interface.
+ *
+ * TODO: add support for devices with dual bank memory, e.g. switch to dual bank
+ * mode, if FLASHSECTOR_2ND_BANK_START is defined and specifies the address of
+ * the first sector in the second bank
+ *
+ * @note        Flash memory has only a limited amount of erase cycles (mostly
+ *              around 10K times), so using this interface in some kind of loops
+ *              can damage you MCU!
+ *
+ * @author      Jannik Beyerstedt <jannik.beyerstedt@haw-hamburg.de>
+ */
+
+#ifndef FLASHSECTOR_H
+#define FLASHSECTOR_H
+
+#include <stdint.h>
+
+#include "periph_cpu.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/* HAL Library macros */
+#define POSITION_VAL(VAL)     (__CLZ(__RBIT(VAL)))
+
+/**
+ * @brief   defines for flash programming parallelism
+ */
+#define FLASH_PSIZE_BYTE           0x00000000U
+#define FLASH_PSIZE_HALF_WORD      0x00000100U
+#define FLASH_PSIZE_WORD           0x00000200U
+#define FLASH_PSIZE_DOUBLE_WORD    0x00000300U
+#define CR_PSIZE_MASK              0xFFFFFCFFU
+
+/**
+ * @brief   flash keys form STM32F4xx
+ */
+#define RDP_KEY                  ((uint16_t)0x00A5)
+#define FLASH_KEY1               0x45670123U
+#define FLASH_KEY2               0xCDEF89ABU
+#define FLASH_OPT_KEY1           0x08192A3BU
+#define FLASH_OPT_KEY2           0x4C5D6E7FU
+
+/**
+ * @brief   Per default, we expect the internal flash to start at address 0
+ */
+#ifndef CPU_FLASH_BASE
+#define CPU_FLASH_BASE      (0)
```

Comment:
I don't think this can be used like that. `CPU_FLASH_BASE` must reflect the real address on which the flash space starts, and AFAIK this is not true for stm32f4 devices. Anyways, I think this will be override [here](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32f4/include/cpu_conf.h#L54).